### PR TITLE
Add Vector PCB to QMK

### PR DESCRIPTION
**Pull request by awkannan**
## Description

Adding Vector keyboard to QMK

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/cannonkeys/vector/readme.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+* Hardware Supported: STM32F072CBT6
+* Hardware Availability: [CannonKeys](https://cannonkeys.com)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make cannonkeys/vector:default
+
+Flashing example for this keyboard:
+
+    make vector:default:flash
```
Comment (keyboard-magpie) [inline keyboards/cannonkeys/vector/readme.md:15]:
```suggestion
    make cannonkeys/vector:default:flash
```

File: keyboards/cannonkeys/vector/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    _FN3
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [_BASE] = LAYOUT_all(
+    KC_GESC, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,                 KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,   KC_BSPC, KC_BSPC,
+    KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,   KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC,  KC_BSLS,
+    KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,                 KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,      KC_ENT,
+    KC_LSFT, KC_NUBS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,                 KC_M,    KC_COMM, KC_DOT,  KC_SLSH,                    KC_RSFT, MO(_FN1),
+    KC_LCTL, KC_LGUI, KC_LALT,                   KC_SPC,  KC_SPC,  KC_SPC,                                 KC_RALT, KC_RGUI, KC_RCTL
```
Comment (keyboard-magpie) [inline keyboards/cannonkeys/vector/keymaps/via/keymap.c:19]:
```suggestion
    KC_GESC, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,   KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,   KC_BSPC, KC_BSPC,
    KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,   KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC,  KC_BSLS,
    KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,      KC_ENT,
    KC_LSFT, KC_NUBS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,    KC_RSFT, MO(_FN1),
    KC_LCTL, KC_LGUI, KC_LALT,                   KC_SPC,  KC_SPC,  KC_SPC,                                 KC_RALT, KC_RGUI, KC_RCTL
```

File: keyboards/cannonkeys/vector/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    _FN3
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [_BASE] = LAYOUT_all(
+    KC_GESC, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,                 KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,   KC_BSPC, KC_BSPC,
+    KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,   KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC,  KC_BSLS,
+    KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,                 KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,      KC_ENT,
+    KC_LSFT, KC_NUBS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,                 KC_M,    KC_COMM, KC_DOT,  KC_SLSH,                    KC_RSFT, MO(_FN1),
+    KC_LCTL, KC_LGUI, KC_LALT,                   KC_SPC,  KC_SPC,  KC_SPC,                                 KC_RALT, KC_RGUI, KC_RCTL
```
Comment (keyboard-magpie) [inline keyboards/cannonkeys/vector/keymaps/default/keymap.c:19]:
```suggestion
    KC_GESC, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,   KC_BSPC, KC_BSPC,
    KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,   KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC,  KC_BSLS,
    KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,      KC_ENT,
    KC_LSFT, KC_NUBS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,    KC_RSFT, MO(_FN1),
    KC_LCTL, KC_LGUI, KC_LALT,                   KC_SPC,  KC_SPC,  KC_SPC,                                 KC_RALT, KC_RGUI, KC_RCTL
```

**Review by keyboard-magpie — CHANGES_REQUESTED** [review]
Minor formatting suggestions.

File: keyboards/cannonkeys/vector/config.h [hunk 4]
```diff
@@ -0,0 +1,32 @@
+// Copyright 2022 Andrew Kannan (@awkannan)
```
Comment (zvecr) [inline keyboards/cannonkeys/vector/config.h:1]: This file can be removed

File: keyboards/cannonkeys/vector/info.json [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true
+    },
+    "matrix_pins": {
+        "cols": ["C13", "C15", "B13", "B12", "B11", "B10", "B2", "B1", "B0", "A7", "A6", "A5", "A4", "A3"],
+        "rows": ["C14", "A1", "B14", "B15", "A8"]
+    },
```
Comment (zvecr) [inline keyboards/cannonkeys/vector/info.json:18]:
```suggestion
    },
    "indicators": {
        "caps_lock": "B3",
        "on_state": 0
    },
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Comment by awkannan** [comment]
@zvecr this one should be good to go now.
Thank you for the callout on moving things into info.json and removing config.h

I merged master in so I could test that it all worked - it does.

**Review by drashna — APPROVED** [review]

**Comment by awkannan** [comment]
Hey @drashna thank you for the approval :)
I just wanted to bump this up a bit as its been a few weeks.

**Review by fauxpark — APPROVED** [review]

**Pull request merged**